ast/src/parsed/mod.rs

```diff
@@ -1476,6 +1476,12 @@ impl<Ref> Children<Expression<Ref>> for ArrayExpression<Ref> {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub enum TypeDeclaration {
```

Comment:
In which way is this TypeDeclaration different from the one above? Can we make it generic over the type of expression?

Reply:
Also why not put this next to StructDeclaration and EnumDeclaration? How is it related to Pattern / ArrayExpression?